core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -479,6 +484,23 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
     getAll.sorted.map{case (k, v) => k + "=" + v}.mkString("\n")
   }
 
+  def getDriverExtraClassPath() : Option[String] = {
+    // Concatenate common and driver classpath in that order
+    val commonExtraClassPath = getOption("spark.common.extraClassPath")
+    val driverExtraClassPath = getOption("spark.driver.extraClassPath")
+    Option((commonExtraClassPath ++ driverExtraClassPath).mkString(File.pathSeparator)).filter(_
```

Comment:
this is super minor, but using `Option` as a wrapper makes the reader think that the you are checking for whether or not there is null inside.

```scala
scala> Option("foo")
res20: Option[String] = Some(foo)

scala> Option("")
res21: Option[String] = Some()

scala> Option(null)
res22: Option[Null] = None
```

but really, you know there won't be any nulls, you are just trying to wrap the String in an option.  The more conventional thing to do in that case is to wrap with `Some`.  The end result is the same in this case -- both do the same thing when the input isn't null -- but the intention is a bit more clear.